### DISCUSSION TITLE
Refactor component.rs rename()

### DISCUSSION THREAD

**Discussion by kellpossible** [juhaku/utoipa#147] [Ideas]
Hi @juhaku  I was attempting to write a documentation comment for `rename()` https://github.com/juhaku/utoipa/blob/a1c49de28cef91d8ba76fb8b42f1c11c9f43287e/utoipa-gen/src/schema/component.rs#L720-L734 and was struggling, starting to think it could be ripe for refactoring. Perhaps we could transform it/it's uses to use less `mut` parameters, and make it easier to reason about what's going in/out in a functional style?

**Comment by kellpossible** [comment]
I guess the point of taking mut is to partly avoid applying the rename more than once, and avoiding taking something by reference by taking it by value from the option? Also with more operations than just rename to come in `Serde` perhaps the rule scope could be reduced to only accept rename rules?

**Comment by kellpossible** [comment]
I put some proposed documentation updates in this draft PR https://github.com/juhaku/utoipa/pull/146/files#diff-c2c506c1d61101323c309d3947b6328198fda433dd1be11232fab56d74345b02R762

**Comment by juhaku** [comment]
Yeah, it indeed is not the easiest function to reason about, and perhaps could be improved a bit.
